src/engine/platform/mixer.h

```diff
@@ -127,21 +147,33 @@ void stop_sound(AudioID id);
 Channel *fetch_channel(u32 channel_id);
 
 ///*
-// Activates delay on the channel with the specified settings.
+// Sets delay on the channel with the specified settings. Unset by
+// setting either feedback or len_seconds to 0.
 void Channel::set_delay(f32 feedback, f32 len_seconds);
 
 ///*
-// De-activates delay on the channel.
-void Channel::remove_delay();
+// Sets target delay on the channel with the specified settings. The feedback
+// and length is changed over time and reaches their targets after delta_seconds.
+void Channel::set_delay_at_time(f32 feedback, f32 len_seconds, f32 delta_seconds);
```

Comment:
These could be merged into one function, with default arguments. So something sane for the delta_seconds, because I doubt anyone actually wants it to take effect right at that time, and if they do you can easily specify the delta to be 0.

Reply:
Good idea